runtime/fastly/host-api/host_api.cpp

```diff
@@ -12,24 +12,86 @@
 
 using api::FastlyResult;
 using fastly::FastlyAPIError;
+using host_api::MonotonicClock;
 using host_api::Result;
 
+#define NEVER_HANDLE 0xFFFFFFFE
+
 size_t api::AsyncTask::select(std::vector<api::AsyncTask *> *tasks) {
   size_t tasks_len = tasks->size();
   fastly_compute_at_edge_async_io_handle_t *handles =
       new fastly_compute_at_edge_async_io_handle_t[tasks_len];
-  for (int i = 0; i < tasks_len; i++) {
-    handles[i] = tasks->at(i)->id();
+  uint64_t now = 0;
```

Comment:
The allocation of `handles` looks like it leaks here, as I'm not finding a corresponding `delete`. What do you think about making it a `std::vector<fastly_compute_at_edge_async_io_handle_t>` instead? You could use the `reserve` method with `tasks_len` to pre-allocate the full buffer, and avoid the manual index tracking with `task_idx` by using `emplace_back` instead. Additionally, you could use the `data()` method to fetch the raw pointer when constructing the `fastly_world_list_handle_t` value lower down.

As there are a lot of early returns in this function, avoiding the need to `delete` the `handles` pointer at each would be good.

Reply:
Added a vector.